vue-app/src/components/FilterDropdown.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+  <div class="category-filter" v-click-outside="closeDropdown">
+    <div class="filter-btn" @click="toggleDropdown">
+      <span class="filter-text">
+        Filter
+        <span v-if="selectedCategories.length"
+          >({{ selectedCategories.length }})</span
+        >
+      </span>
+      <img
+        src="@/assets/chevron-down.svg"
+        alt="Down arrow"
+        :class="{ 'filter-chevron-open': visible }"
+      />
+    </div>
+    <div class="selector-wrapper" :class="{ show: visible }">
+      <div
+        v-for="(category, idx) of categories"
+        :key="idx"
+        :class="{
+          'category-btn': true,
+          'category-btn-selected': selectedCategories.includes(category),
+        }"
+        @click="handleFilterClick(category)"
+      >
+        {{ category }}
+        <img
+          v-if="selectedCategories.includes(category)"
+          class="close"
+          src="@/assets/close.svg"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import Component from 'vue-class-component'
+import { Prop } from 'vue-property-decorator'
+
+import ClickOutside from '@/directives/ClickOutside'
+
+@Component({
+  directives: {
+    ClickOutside,
+  },
+})
+export default class FilterDropdown extends Vue {
+  @Prop({ default: [] }) categories!: string[]
+  @Prop({ default: [] }) selectedCategories!: string[]
+
+  private visible = false
+
+  toggleDropdown(): void {
+    this.visible = !this.visible
+  }
+
+  closeDropdown(event): void {
+    if (!event.target.matches('.close')) {
+      this.visible = false
+    }
+  }
+
+  handleFilterClick(selection: string): void {
+    this.$emit('change', selection)
+  }
+}
+</script>
+
+<style scoped lang="scss">
+@import '../styles/theme';
+@import '../styles/vars';
+
+.category-filter {
+  width: 140px;
+  grid-area: filter;
+  position: relative;
+
+  .filter-btn {
+    background: none;
+    cursor: pointer;
+    border: 1px solid $border-color;
+    border-radius: 0.75rem;
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    gap: 1rem;
+    padding: 0.5rem 0.75rem;
+    transition: transform 0.1s;
+    &:hover {
+      opacity: 0.8;
+      transform: scale(1.01);
+    }
+    .filter-chevron-open {
+      transform: rotate(180deg);
+    }
+    @media (max-width: $breakpoint-m) {
+      width: auto;
+    }
+  }
+
+  .selector-wrapper {
+    display: none;
+    position: absolute;
+    top: 110%;
+    right: 0;
+    grid-template-columns: repeat(1, 4fr);
+    border: 1px solid $border-color;
+    border-radius: 0.75rem;
+    overflow: hidden;
+    justify-content: space-between;
+    width: 100%;
+    z-index: 99;
+    .category-btn {
+      display: grid;
+      place-items: center;
```

Comment:
<img width="208" alt="Screen Shot 2021-06-24 at 3 25 31 PM" src="https://user-images.githubusercontent.com/15589226/123335498-c911c280-d501-11eb-99f0-6f1597690bd2.png">
<img width="174" alt="Screen Shot 2021-06-24 at 3 25 39 PM" src="https://user-images.githubusercontent.com/15589226/123335502-c9aa5900-d501-11eb-9404-530726c2a9da.png">

Just wanted to show these side by side where we change the style of `place-items: center;` to `place-items: left;` and see what you guys thought. Its a bit nit picky, just was looking at it in comparison after @pettinarip changes. Im dont have a strong opinion, but the consistency of `place-items: left;` may be nice if .category-btn-selected has `justify-content: space-between;` as the style.

Reply:
True! I hadn't realized that. Yes, I agree that this should be left aligned.

Reply:
@samajammin what do you think?

Reply:
Didn't even spot this at first, but good catch @corwintines. Given that the "selected" state displays much better left aligned (with the close button right aligned), I think having the "unselected" items default to left align makes more sense. 

Otherwise this all looks and functions great on my end! Nice job @pettinarip!

Reply:
Agreed - let's display these categories left aligned.

Reply:
Done!